### ISSUE TITLE
Keyboard a11y -  ESC and ALT-UP keys don't close autocomplete

### ISSUE THREAD

**Issue by DaveBest99** [angular/material2#3378]
#### Bug, feature request, or proposal:
Bug: Keyboard user cannot close the list box with Escape or alt+up_arrow keys.

#### What is the expected behavior?
For an auto-complete that has text in the text box that was both typed by the user and auto-completed by highlighting a choice using the keyboard, pressing escape or alt+up_arrow closes the drop-down list and should clear the text box. Pressing alt+down_arrow should open the list box and allow the user to press up/down arrow keys in the list box and type in the Edit box.

#### What is the current behavior?
Currently the keyboard user cannot close the list box without selecting an item.

#### What is the use-case or motivation for changing an existing behavior?
To allow keyboard users a good interactive experience.

#### Is there anything else we should know?
Pressing the Escape key appears to do nothing. Pressing the Alt+up_arrow JAWS says "close list box", but nothing actually happens as the user can still press up/down arrow keys through the list box. The Edit box is not cleared. Pressing alt+down_arrow has the same effect as pressing down arrow. The alt+up_arrow should collapse the list box and alt+down_arrow should expand the list box.

**Comment by kara** [comment]
Related: https://github.com/angular/material2/issues/3416

**Comment by crisbeto** [comment]
Closing both through escape and alt+up arrow is supported now.
